packages/runtime/src/runtimes/react/controls/control.tsx

```diff
@@ -61,14 +76,13 @@ export function ControlValue({
     case SelectDefinition.type:
     case IconRadioGroupDefinition.type:
     case ComboboxDefinition.type:
-      return (
-        <RenderHook
-          key={definition.controlType}
-          hook={useResolvedValue}
-          parameters={[data, definition]}
-        >
+      const resolved = resolveValue(data, definition)
```

Comment:
Can we call `resolveValue` like this on render and expect proper subscriptions, etc. to happen resulting in re-renders?

Reply:
Great question, forgot to comment on this, this only works because so far we've merged only a "half-way" version of `resolveValue` that is effectively a straight up data transformation call. There are no subscriptions or side-effects here. The proper `useResolvedProps` hook will have to tackle this differently (or maybe fallback to the error boundary approach after all).